pkg/git/git.go

```diff
@@ -244,3 +249,35 @@ func userHasKnownHostsFile(logger *zap.SugaredLogger) (bool, error) {
 	f.Close()
 	return true, nil
 }
+
+func validateGitAuth(logger *zap.SugaredLogger, url string) error {
+	homepath, err := homedir.Dir()
+	if err != nil {
+		logger.Errorf("Unexpected error: getting the user home directory: %v", err)
+		return err
+	}
+	euid := os.Geteuid()
+	sshCred := true
+	// Special case the root user/directory
+	if euid == 0 {
+		if _, err := os.Stat("/root/.ssh"); os.IsNotExist(err) {
+			sshCred = false
+		}
+	} else {
+		if _, err := os.Stat(homepath + "/.ssh"); os.IsNotExist(err) {
+			sshCred = false
+		}
+	}
+	sshFormatRegex := regexp.MustCompile(sshURLFormat)
```

Comment:
Given the format remains fixed we can probably compile this once at the top of the file? e.g.

```go
// near top of file
var sshFormatRegex = regexp.MustCompile(sshURLFormat)
```